webui/react/src/pages/TrialDetails/TrialDetailsWorkloads.tsx

```diff
@@ -106,8 +110,52 @@ const TrialDetailsWorkloads: React.FC<Props> = ({
     });
   }, [ metrics, settings, trial, experiment ]);
 
+  const [ workloads, setWorkloads ] = useState<WorkloadGroup[]>([]);
+  const [ workloadCount, setWorkloadCount ] = useState<number>(0);
+
+  const fetchWorkloads = useCallback(async () => {
+    try {
+      if (trial?.id) {
+        const wl = await getTrialWorkloads({
+          filter: settings.filter,
+          id: trial.id,
+          limit: settings.tableLimit,
+          offset: settings.tableOffset,
+          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+          sortKey: settings.sortKey,
+        });
+        setWorkloads(wl.workloads);
+        setWorkloadCount(wl.count);
+      } else {
+        setWorkloads([]);
+        setWorkloadCount(0);
+      }
+    } catch (e) {
+      handleError(e, {
+        publicMessage: 'Failed to load recent trial workloads.',
+        publicSubject: 'Unable to fetch Trial Workloads.',
+        silent: false,
+        type: ErrorType.Api,
+      });
+    }
+  }, [
+    trial?.id,
+    settings.sortDesc,
+    settings.sortKey,
+    settings.tableLimit,
+    settings.tableOffset,
+    settings.filter,
+  ]);
+
+  // const { stopPolling } =
+  usePolling(fetchWorkloads);
```

Comment:
if you want to rerun the polling function when the params change, `rerunOnNewFn` will do that (for freshness of data)